newsletter2go/newsletter2go.php

```diff
@@ -70,9 +84,70 @@ public function install()
         return parent::install()
             && $this->registerUrls()
             && $this->registerHook('backOfficeHeader')
-            && $this->registerHook('displayOrderConfirmation');
+            && $this->registerHook('displayOrderConfirmation')
+            && $this->registerHook('actionCartSave');
+    }
+
+    /*
+     * this hook will be triggered when:
+     * - a product is added to the cart
+     * - product amount is increased or decreased
+     * - a product is deleted
+     * @param $params
+     */
+    public function hookActionCartSave($params)
+    {
+        if (Configuration::get('NEWSLETTER2GO_ABANDONED_SHOPPING_CART') === '1') {
+            $cart = $params['cart'];
+
+            if (isset($cart)) {
+                $shop = $this->context->shop->getShop($cart->id_shop);
+                $productData = [];
+
+                foreach ($cart->getProducts(true) as $products) {
+                    $product[] = [
+                        'id' => (string)$products['id_product'],
+                        'quantity' => (string)$products['quantity']
+                    ];
+
+                    $productData = array_merge($productData, $product);
+                }
+
+                if(isset($cart->id_customer) && $cart->id_customer != 0){
```

Comment:
if this is false, `$customer` will be undefined and it is used in the `$cartData`.This can make problems depending on some shops settings